docs/components/input.md

```diff
@@ -159,6 +159,24 @@ String to be displayed as error message if `hasError` prop is `true`.
 <KInput label="Large" size="large" hasError errorMessage="Service name should not contain '_'" :overlay-label="true" />
 ```
 
+### iconPosition
+
+Controls position of the icon provided through the [slot](#slots). Accepted values: `left` (default) and `right`.
```

Comment:
suggestion: I'm not a big fan of `right`/`left` - if we ever get big enough to support a RTL language this will not translate well. Can we use `leading`/`trailing` instead?

Reply:
Even if we support RTL, `left/right` still makes sense as it's the position of the icon, not the "start text on the left/right"

Reply:
I think so too ^

Reply:
That's not how RTL rendering works. It's explained in detail [here](https://medium.com/@ahmed044/leading-trailing-vs-left-right-constraints-c315e6f92aa9#:~:text=In%20locales%20where%20the%20reading%20direction%20is%20right%20to%20left,will%20be%20the%20left%20side.), but I'll give a simple example.

I have a use case to have a search input with a clear button rendered at the end.

In LTR I would render that like this:

```html
<KInput icon-position="right">
  <template #icon>
    <KIcon icon="clear" />
  </template>
</KInput>
```

in RTL I would render it this way:

```html
<KInput icon-position="left">
  <template #icon>
    <KIcon icon="clear" />
  </template>
</KInput>
```

So if I want to support both languages now I have to introduce a conditional. But instead I am suggesting we use `leading`/`trailing` - in that scenario I don't need a conditional, this code does what I need in both scenarios.

```html
<KInput icon-position="trailing">
  <template #icon>
    <KIcon icon="clear" />
  </template>
</KInput>
```

Reply:
We decided to go with `start`/`end`

Reply:
I'm still not sure this makes sense to me as the component still doesn't behave how you're suggesting when I manually change the direction, meaning there would have to be conditional logic regardless.

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/2229946/224421146-86e7fa4d-18ca-44ce-830f-1e7cd5d21081.png">

- The first input above has no modifications, and you can see the icon is displayed on the left.
- For the second input wrapper I have manually changed the `direction: rtl` and you can see the input text _does_ switch as expected; however, the icon position remains on the left.

Reply:
That's true - we paired with Kai and I asked the same question. Basically what I got from him is that yes, to fully support RTL we would have to redefine bunch of styles (in this case too). However, if we supported RTL and still had `left` and `right` values accepted by this prop, one would have to leave them as is, which would result in host app having to introduce a condition (something like `:icon-position="isRtl? 'left' : 'right'"`). I agreed with Kai that that would be confusing. @kaiarrowood anything to add?

Reply:
Don't mean to be rude but Adam should be enjoying his Friday off 😅

Reply:
We'd have to do a bunch of changes to support RTL anyway; whatever you all think is good by me. I was just noting that the behavior is still not there regardless of the prop name

Reply:
Yeah, the point of changing the prop values was not to achieve certain behaviour, but rather to abstract towards more relative values

Reply:
Yeah, my thought process is that the logic to correctly display an element in RTL should live in the component and not the consuming application. If we use left/right the host application is forced to conditionally apply the treatment because when speaking about RTL support, the vast majority of use cases want things positioned at the start/end of something rather than always displaying on the left or always displaying on the right. Just speaking from experience I haven't really been able to think of a use case and if you go to a website that supports RTL and switch to an RTL language you'll see the same thing - which is that the UI is basically mirrored, not just the text on the page.